adb_shell/adb_device.py

```diff
@@ -425,6 +458,30 @@ def shell(self, command, timeout_s=None, total_timeout_s=constants.DEFAULT_TOTAL
         """
         return self._service(b'shell', command.encode('utf8'), timeout_s, total_timeout_s, decode)
 
+    def streaming_shell(self, command, timeout_s=None, total_timeout_s=constants.DEFAULT_TOTAL_TIMEOUT_S, decode=True):
+        """Send an ADB shell command to the device.
```

Comment:
Similar to the above comment, please add something to this method's description to distinguish it from _service. How about:

Send an ADB shell command to the device, yielding each line of output.